ui/src/public/workspace.ts

```diff
@@ -216,6 +216,20 @@ export class TrackNode {
     }
   }
 
+  /**
+   * Get all ancestors of this node from root to immediate parent.
+   * Returns an empty array if this node has no parent.
+   */
+  getAncestors(): TrackNode[] {
+    const ancestors: TrackNode[] = [];
+    let current = this.parent;
+    while (current && current.name !== '') {
```

Comment:
Why are we stopping at nodes with blank names?

Reply:
Because blank names = root nodes. We don't want to include the workspace nade here.